v0.7.2

danamlewis opened the pull request:
Setting up for bug fixes and other updated for v0.7.2

The primary fix in this release is for:
* #1435 - updates to golang broke new installations.

Other fixes/improvements include:
* #1429
* #1419 various node/npm improvements
* fix for #1436
* #1443 to use compression when sending data to NS
* #1438 convert_bg (for mmol/L users) in console.error logs
* #1457 improving pre-flight for multiple rig environments

TODO before release: 

- [ ] Revert https://github.com/openaps/oref0/pull/1433/commits/21620183a582c395bd2255a0458684b9a93a6651

scottleibrand committed:
0.7.1

dcacklam committed:
Support for Debian Bullseye and aarch64 (Raspberry Pi OS/Bullseye 64 Bit) in setup routines (#1429)

* Remove redundant pip install for bullseye support

pIp is also installed by openaps-packages.sh

* Change python default to python2, install pip, pip install numpy for bullseye support

* Bullseye support on arm64 (Raspberry Pi OS 64bit)

Handle aarch64 architecture properly when downloading golang binaries. 
Without this setup will fail on Rpi 3/4/Zero2 devices where the user has installed the recently-released 64bit OS.

Co-authored-by: Dave Acklam <dave.acklam@zonarsystems.com>

mountrcg committed:
omit string formating in convert_bg (#1414)

fixes #1413

ChanceHarrison committed:
Improved handling of node installation (& miscellaneous improvements) (#1419)

* fix: Remove edison nodesource installation

* fix: npm upgrade to latest *compatible* version, not the literal latest

The previous command (which installs npm@latest) results in the installation
of a version of npm which is almost certainly incompatible with the installed
(regardless of source) version of nodejs

* fix: Use n instead of apt for installing node & npm

* fix: Improve node execution time check function

- Remove what should be a redundant install, instead opt to fail if node is missing.
Given that openaps-packages.sh handles installing node, this failing check indicates
an unmitigated failure in the process previously, so let's not continue and make things
potentially worse.
- Attempt to remove existing nodejs/npm packages using apt
- Warn the user if versions of node and npm remain
- Switch nvm for n, using installed version of n if available
Else, bootstrap install n as done elsewhere.

* fix: Replace broken grep statements

scottleibrand committed:
TODO: revert before release; use dev/bin/openaps-packages.sh

scottleibrand committed:
check out the specified branch when cloning

scottleibrand committed:
use golangversion=1.19.1 and go install to fix #1435

scottleibrand committed:
symlink Go-mmtune to point to oref0-mmtune

scottleibrand committed:
Revert "symlink Go-mmtune to point to oref0-mmtune"

This reverts commit 84f10113e8a87c7abc60cc34ba2bcc6c61e8441e.

scottleibrand committed:
install mmtune

scottleibrand committed:
symlink /usr/local/bin/Go-mmtune to point to /root/go/bin/mmtune

scottleibrand committed:
have to go install each medtronic command individually

scottleibrand committed:
go install puts things in /root/go/bin/

scottleibrand committed:
have to comment out if clause if it has no uncommented contents

scottleibrand committed:
use ... to install all medtronic commands

scottleibrand committed:
don't die if Go-mmtune symlink already exists

scottleibrand committed:
fix path to openaps.jq

scottleibrand commented on bin/oref0-setup.sh:
```diff
@@ -1269,7 +1269,7 @@ if prompt_yn "" N; then
         #go get -u -v -tags "$radiotags" github.com/ecc1/medtronic/... || die "Couldn't go get medtronic"
         go install -v -tags "$radiotags" github.com/ecc1/medtronic/cmd/...@latest || die "Couldn't go get medtronic"
-        ln -sf $HOME/go/src/github.com/ecc1/medtronic/cmd/pumphistory/openaps.jq $directory/ || die "Couldn't softlink openaps.jq"
+        ln -sf /root/go/pkg/mod/github.com/ecc1/medtronic@v0.0.0-20210712211734-b8431dc5211b/cmd/pumphistory/openaps.jq $directory/ || die "Couldn't softlink openaps.jq"
```
@ecc1 This works, but looks fragile. What's the proper way to do this now that `go get` is deprecated?

ecc1 commented:
I guess I'd use wget to fetch a copy from GitHub directly.

On Sun, Sep 18, 2022, 4:31 AM Scott Leibrand ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In bin/oref0-setup.sh
> <https://github.com/openaps/oref0/pull/1433#discussion_r973682317>:
>
> > @@ -1269,7 +1269,7 @@ if prompt_yn "" N; then
>          #Build Go binaries
>          #go get -u -v -tags "$radiotags" github.com/ecc1/medtronic/... || die "Couldn't go get medtronic"
>          go install -v -tags "$radiotags" ***@***.*** || die "Couldn't go get medtronic"
> -        ln -sf $HOME/go/src/github.com/ecc1/medtronic/cmd/pumphistory/openaps.jq $directory/ || die "Couldn't softlink openaps.jq"
> +        ln -sf ***@***.***/cmd/pumphistory/openaps.jq $directory/ || die "Couldn't softlink openaps.jq"
>
> @ecc1 <https://github.com/ecc1> This works, but looks fragile. What's the
> proper way to do this now that go get is deprecated?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openaps/oref0/pull/1433#pullrequestreview-1111508343>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALZHRO74EORHEPFGOD2DMTV63HPNANCNFSM5ZFUPWJA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

scottleibrand committed:
wget openaps.jq instead of symlinking to the 20210712211734 version

Commit:
fix(lib/iob): Move value checks up to index.js

scottleibrand committed:
Merge branch 'ChanceHarrison-chance/fix-1436' (#1437) into dev

tzachi-dar committed:
Use compression for operations that go to nightscout. (#1443)

mountrcg committed:
convert_bg missing in many statements for rT (#1438)

This commit calls convert_bg on various console.error() print statements, which should improve readability for mmol/L users and leave mg/dL users' experience unaffected.

scottleibrand commented:
This should be ready for release now, to fix installs from master, unless anyone knows of any blockers.

scottleibrand committed:
Revert "fix(lib/iob): Move value checks up to index.js"

This reverts commit 80493da410ff9c102c02794c08a706af02fabc3b.

This is causing future IOB predictions to be incorrect (highly negative).
That causes the predBG lines to swoop up and to the right.
That results in (slightly) too much insulin being dosed in some situations.

scottleibrand committed:
#1455: Don’t cancel a high temp due to lack of BG data if the temp has been running for less than 10m

scottleibrand committed:
don't modify temp if lastTempAge < 10, just print why not

scottleibrand committed:
remove suprious BOM introduced in #1443 causing /usr/bin/env: No such file

scottleibrand committed:
add ', but' to make reason more readable

scottleibrand committed:
if iob_data.lastTemp is undefined, calculate lastTempAge from currenttemp.duration % 30

scottleibrand committed:
wait upto45s and try preflight; if successful, refresh pumphistory, else mmtune (#1457)

* on loop failure, retry preflight; if successful, refresh pumphistory, else mmtune

* wait_for_silence  before retrying preflight

* need to return, not keep invoking fail()

* only try preflight one more time before mmtuning

scottleibrand committed:
use archive.debian.org for stretch now that it's EOL

scottleibrand committed:
IPv6 should be fine, but let's use defaults to preserve conf files

hpeuscher committed:
fix missing declaration

inventor96 committed:
Fixed NS perms checking

inventor96 committed:
Fixed NS perms checking

inventor96 committed:
Removed duplicate line

inventor96 committed:
Removed duplicate line

inventor96 committed:
Revert scope promotion

Jon-b-m committed:
rT.reason missing colons (:)

In some places a space is used instead of a colon.

scottleibrand committed:
Fix Stretch archived apt installs

Rewrite Stretch sources to archive.debian.org only, remove dead Stretch entries, and add archive apt settings so later apt-get install steps succeed on archived repos.
